lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -71,6 +71,10 @@ public function get_items_permissions_check( $request ) {
 
 		$post_type = get_post_type_object( $this->post_type );
 
+		if ( ! empty( $request['orderby'] ) && $request['orderby'] === 'relevance' && empty( $request['search'] ) ) {
```

Comment:
@websupporter I think this should be in `get_items` as this is a logic check, not to do with permissions.